cms/release_calendar/tests/test_models.py

```diff
@@ -0,0 +1,187 @@
+from cms.release_calendar.models import ReleasePageRelatedLink
+
+pytestmark = pytest.mark.django_db
+
+
+@pytest.mark.parametrize(
+    "status,suffix",
+    [
+        (ReleaseStatus.PROVISIONAL, "--provisional.html"),
+        (ReleaseStatus.CONFIRMED, "--confirmed.html"),
+        (ReleaseStatus.CANCELLED, "--cancelled.html"),
+        (ReleaseStatus.PUBLISHED, ".html"),
+    ],
+)
+def test_release_calendar_page_template(rf, release_calendar_page, status, suffix):
+    """Check the template used."""
+    request = rf.get("/")
+    release_calendar_page.status = status
+    assert (
+        release_calendar_page.get_template(request)
+        == f"templates/pages/release_calendar/release_calendar_page{ suffix }"
+    )
+
+
+def test_release_calendar_page_related_links_for_context(rf, release_calendar_page):
+    """Check that the correct related links are passed to the context."""
+    request = rf.get("/")
+    related_link = ReleasePageRelatedLink(
+        parent=release_calendar_page, link_url="https://ons.gov.uk", link_text="The link"
+    )
+    related_link.save()
+
+    expected = [
+        {"text": "The link", "url": "https://ons.gov.uk"},
+    ]
+    assert release_calendar_page.related_links_for_context == expected
+    assert release_calendar_page.get_context(request)["related_links"] == expected
+
+
+@pytest.mark.parametrize(
+    "status",
+    [
+        ReleaseStatus.PROVISIONAL,
+        ReleaseStatus.CONFIRMED,
+        ReleaseStatus.CANCELLED,
+    ],
+)
+def test_release_calendar_page_table_of_contents_pre_published(rf, release_calendar_page, status):
+    """Check TOC in a pre-published state."""
+    request = rf.get("/")
+    release_calendar_page.status = status
+    release_calendar_page.content = [
+        {
+            "type": "release_content",
+            "value": {"title": "Publications", "links": [{"external_url": "https://ons.gov.uk", "title": "test"}]},
+        }
+    ]
+
+    related_link = ReleasePageRelatedLink(
+        parent=release_calendar_page, link_url="https://ons.gov.uk", link_text="The link"
+    )
+    related_link.save()
+
+    expected = [{"url": "#summary", "text": "Summary"}]
+    assert release_calendar_page.table_of_contents == expected
+    assert release_calendar_page.get_context(request)["table_of_contents"] == expected
+
+    expected_with_census_or_accredited = [*expected, {"url": "#about-the-data", "text": "About the data"}]
+
+    del release_calendar_page.table_of_contents  # clear the cached property
+    release_calendar_page.is_census = True
+    release_calendar_page.is_accredited = False
+
+    assert release_calendar_page.table_of_contents == expected_with_census_or_accredited
+    assert release_calendar_page.get_context(request)["table_of_contents"] == expected_with_census_or_accredited
+
+    del release_calendar_page.table_of_contents  # clear the cached property
+    release_calendar_page.is_census = False
+    release_calendar_page.is_accredited = True
+
```

Comment:
Suggestion: This feels like it's testing a number of different conditions in the same test. It's probably worth sticking to ["Arrange Act Assert"](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/) where possible.